### ISSUE TITLE
request: RemoteTech_XF

### ISSUE THREAD

**Issue by abenkovskii** [KSP-CKAN/CKAN-meta#351]
http://forum.kerbalspaceprogram.com/threads/108669-0-90-RemoteTech-XF-RemoteTech-v-1-6-3

**Comment by Dazpoet** [comment]
This mod requires the ability to overwrite files owned by another mod, in this case RemoteTech. CKAN cannot overwrite files owned by another mod since it breaks the consistency rules the program is built on. Therefore I will close this issue since we cannot add this mod to the repository.

**Comment by abenkovskii** [comment]
@Dazpoet Can we have two versions of RT? Other mods like ActiveTextureManagement do so.

**Comment by Dazpoet** [comment]
Based on this [post](http://forum.kerbalspaceprogram.com/threads/108669-0-90-RemoteTech-XF-RemoteTech-v-1-6-3?p=1696376&viewfull=1#post1696376) in the RemoteTech_XF thread I get the impression that releasing another version bundling the modified .dll with the original assets might not be possible seeing as the author isn't doing so. CKAN does not support downloading assets from one binary and the .dll from another one unless they're broken into different mods.

ATM packages are all released by the same author and as such different versions is an easy thing to do since they are all officiall binaries and can share assets anyhow they want to.

An packaging solution to resolve this issue could be:

- RemoteTech-Core - The original .dll for RemoteTech
- RemoteTech - Everything in the current RemoteTech package except the .dll
- RemoteTech-XF - Package that provides and conflicts with RemoteTech-Core and supplies the modified .dll file

This kind of splitting brings with it some problems we're currently working with in https://github.com/KSP-CKAN/NetKAN/pull/889 but once those are resolved it should be possible. However I'm not sure about the netiquette of these things so I'd like a second opinion @hakan42 @pjf @Ippo343.

**Comment by pjf** [comment]
Remote tech unfortunately has non free models. One possible solution here would be to break those models into a separate package. In an ideal world this would mean remotetech depends on remotetech-models, and remotetech-xf does the same. (And then in my most ideal world, we'd have someone create free versions of all the models.)

Having said that, I'm on my phone and can't easily check how xf is packaged just this moment; there may be other ways we can split this that reflect reality rather than idealism.

**Comment by Dazpoet** [comment]
XF is just a single RemoteTech.dll, along with a readme and license. Installation instructions for XF is to download RemoteTech and then overwrite the vanilla .dll file.

**Comment by Acecool** [comment]
So split it up into Core DLL, Core Files, and XF DLL... So you download the core files along with either Core DLL OR XF DLL... I still think managing these ckan files manually is a huge task.. It should be automated.

**Comment by dbent** [comment]
> I still think managing these ckan files manually is a huge task.. It should be automated.

@Acecool  I'm not sure what you mean. Anything that can be automated already is automated via NetKAN. If the data is being managed manually it's either because it's not technically feasible to automate or because we can't because of terms of service of the hosting site (like Curse).

**Comment by plague006** [comment]
This can be/will be unblocked by https://github.com/KSP-CKAN/CKAN/issues/1311

**Comment by plague006** [comment]
Closing in favour of https://github.com/KSP-CKAN/NetKAN/issues/2179
